third-party/rsutils/src/special-folder.cpp

```diff
@@ -0,0 +1,139 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+#include <rsutils/os/special-folder.h>
+#include <rsutils/easylogging/easyloggingpp.h>
+
+#ifdef _WIN32
+
+#include <rsutils/os/hresult.h>
+#include <KnownFolders.h>
+#include <shlobj.h>
+
+#elif defined __linux__ || defined __APPLE__
+
+#include <unistd.h>
+#include <sys/types.h>
+#include <pwd.h>
+
+#endif
+
+
+namespace rsutils {
+namespace os {
+
+
+std::string get_special_folder( special_folder f )
```

Comment:
Rename `get_special_folder_path`, it does not return a descriptor or such like.

Reply:
It returns a "folder". I don't know if I understand the connection to a descriptor.

Reply:
It returns a string that represents the folder path in the file system. It does not return a folder "object".

Reply:
But there is no such object. Why does it have to be an object? A folder is represented by a path which is represented by a string, almost universally.